Core/src/org/sleuthkit/autopsy/directorytree/ExtractUnallocAction.java

```diff
@@ -170,7 +171,9 @@ public void approveSelection() {
         try {
             List<Content> unallocFiles = c.getChildren();
             if (null != unallocFiles && unallocFiles.isEmpty() == false) {
-                return unallocFiles.get(0).accept(uv); //Launching it on the root directory
+                if (unallocFiles.get(0) instanceof AbstractContent) {
+                    return unallocFiles.get(0).accept(uv); //Launching it on the root directory
+                }
```

Comment:
I find the name "unallocFiles" for the List<Content> and the comment about launching on the root directory confusing here. We are not getting a collection of unallocated files, we are getting the children of an Image or Volume, and making the assumption that the first child is  a root directory to be searched for unallocated space files. Let's document and button this down better. We need a more accurate name for unallocFiles (Maybe just contentChildren?  I'm open to a better idea.)  Then are we sure that the first child of an Image or Volume will always be the "root directory"? Are there overrides of getChildren for Image and Volume that ensure this? It looks like AbstractFile has an isRoot method, is that applicable here? Or, looking at the UnallocVisitor, is the comment just wrong?

Reply:
The getChildren methods for both Image and Volume rely on getChildrenInfo which performs a query whose results will ordered by tsk_objects.obj_id so the first result will be the child which was added to the database first.  Whether or not that will always be a "root directory" I need to investigate the creation order of the children a bit, but it seems reasonable that if there is a root directory being created as a child it will be the first child created.